### PR TITLE
take optional Opts for modifying the http server

### PR DESCRIPTION
**Pull request by rybit**
fixes https://github.com/netlify/netlify-commons/issues/177

This should also be extensible for other settings as they come up

### CODE REVIEW

**Review by mraerino — COMMENTED** [review]
looks like we're now mixing two ways to configure the server:
- use the `Config` struct
- use the opts

I'd like to use one or the other not both.
Generally i'm in favor of the opts approach it's close to a builder pattern, but more flexible in Go.

Maybe consider making a different constructor that doesn't take `Config`, but mostly opts and people can decide which they want to use?

**Comment by rybit** [comment]
Good idea - I'll do that. I was thinking that things in Config would be considered "required" and things in "Opts" would be additional modifications on top of that.

**Comment by mraerino** [comment]
yeah, was thinking about that. since in general we should try to make less things required and provide more opinionated defaults we should try to make all required things explicit parameters imho

**Comment by rybit** [comment]
I did something where it kinda unified them a bit. One pure Opts and one with config. I am not in love with it :/

File: server/server.go [hunk 1]
```diff
@@ -52,44 +56,30 @@ type HealthChecker interface {
 	Healthy(w http.ResponseWriter, r *http.Request) error
 }
 
-func New(log logrus.FieldLogger, config Config, api APIDefinition) (*Server, error) {
-	var healthHandler router.APIHandler
-	if checker, ok := api.(HealthChecker); ok {
-		healthHandler = checker.Healthy
+func NewOpts(log logrus.FieldLogger, api APIDefinition, opts ...Opt) (*Server, error) {
```
Comment (keiko713) [inline server/server.go:59]: THIS IS YOUR TIME TO SHINE AND PUT THE COMMENT HERE, RYAN

**Review by mraerino — DISMISSED** [review]
doesn't look too bad to me 👍

**Review by mraerino — APPROVED** [review]

**Pull request merged**